test/sys/test_uio.rs

```diff
@@ -190,3 +190,51 @@ fn test_preadv() {
     let all = buffers.concat();
     assert_eq!(all, expected);
 }
+
+#[test]
+#[cfg(any(target_os = "linux", target_os = "android"))]
+// FIXME: qemu-user doesn't implement process_vm_readv/writev on most arches
+#[cfg_attr(not(any(target_arch = "x86", target_arch = "x86_64")), ignore)]
+fn test_process_vm_readv() {
+    use nix::unistd::ForkResult::*;
+    use nix::sys::signal::*;
+    use nix::sys::wait::*;
+    use std::{str, slice};
+
+    let (r, w) = pipe().unwrap();
+    match fork() {
```

Comment:
The child calls `pause()`, which means it never continues execution past that point, so it doesn't unwind anything and shouldn't cause any deadlocks in execution. If you're having trouble with deadlocks during child _creation_ then yeah that's a problem but also that seems surprising... I guess another thread might be holding the malloc lock or something? Blah.

I can probably use `process_vm_readv(getpid()...)` with a static string.